src/utils/config.js

```diff
@@ -57,24 +59,30 @@ const attemptToCreateFile = path => {
 	return attemptCallWithWarning(fn, path);
 };
 
-const checkReadAccess = path => {
-	const fn = fs.accessSync.bind(null, path, fs.constants.R_OK);
-	const errorCode = 1;
-	const errorMessage = `Could not read config file. Please check permissions for ${path} or delete the file so we can create a new one from defaults.`;
-	return attemptCallWithError(fn, errorCode, errorMessage);
-};
-
 const checkLockfile = path => {
 	const fn = lockfile.lockSync.bind(null, path);
-	const errorCode = 3;
+	const errorCode = 2;
 	const errorMessage = `Config lockfile at ${lockfilePath} found. Are you running Lisky in another process?`;
 	return attemptCallWithError(fn, errorCode, errorMessage);
 };
 
 const attemptToReadJSONFile = path => {
 	const fn = readJSONSync.bind(null, path);
-	const errorCode = 2;
-	const errorMessage = `Config file is not valid JSON. Please check ${path} or delete the file so we can create a new one from defaults.`;
+	const errorCode = 1;
+	const errorMessage = `Config file cannot be read or is not valid JSON. Please check ${path} or delete the file so we can create a new one from defaults.`;
+	return attemptCallWithError(fn, errorCode, errorMessage);
+};
+
+const attemptToValidateConfig = (config, path) => {
+	const rootKeys = CONFIG_VARIABLES.map(key => key.split('.')[0]);
+	const fn = () =>
+		rootKeys.forEach(key => {
+			if (!Object.keys(config).includes(key)) {
+				throw new ValidationError(`Key ${key} not found in config file.`);
+			}
+		});
+	const errorCode = 3;
```

Comment:
what are these errorCode based on? maybe it's better to define it, or follow standards?

Reply:
It was just incrementing from 1 every time a new error type was needed in this file. I'm very up for following an existing standard, but the natural choice seems to be https://nodejs.org/api/errors.html#errors_node_js_error_codes, but it's unclear if they have codes for all our situations. Do you know of a good alternative?

Reply:
I think the node error code is good although im not sure if it represents number tho.
Maybe something like
[https://shapeshed.com/unix-exit-codes/](https://shapeshed.com/unix-exit-codes/)
Even if we don't follow the standard, we should probably have some meaning behind the number.